lighthouse-core/gather/driver/execution-context.js

```diff
@@ -164,14 +165,79 @@ class ExecutionContext {
    * @return {FlattenedPromise<R>}
    */
   evaluate(mainFn, options) {
-    const argsSerialized = options.args.map(arg => JSON.stringify(arg)).join(',');
+    const argsSerialized = ExecutionContext.serializeArguments(options.args);
     const depsSerialized = options.deps ? options.deps.join('\n') : '';
     const expression = `(() => {
       ${depsSerialized}
       return (${mainFn})(${argsSerialized});
     })()`;
     return this.evaluateAsync(expression, options);
   }
+
+  /**
+   * Evaluate a function on every new frame from now on.
+   * @template {any[]} T
```

Comment:
```suggestion
   * @template {unknown[]} T
```
More or less a nit, but we should start defaulting to `unknown` unless `any` is required. It's a better-behaved top type, doesn't generally spread without noticing it happening, and it's usually more expressive of what is meant ("it doesn't matter what this is") without the secondary meaning of `any` ("my types/tsc are broken, just let me out, please" :)

Reply:
sg